icu4c/source/common/dictbe.cpp

```diff
@@ -1063,6 +1075,7 @@ CjkBreakEngine::CjkBreakEngine(DictionaryMatcher *adoptDictionary, LanguageType
         UnicodeSet cjSet(UnicodeString(u"[[:Han:][:Hiragana:][:Katakana:]\\u30fc\\uff70\\uff9e\\uff9f]"), status);
         if (U_SUCCESS(status)) {
             setCharacters(cjSet);
+            initJapanesePhraseParameter(status);
```

Comment:
This should only be called for -lw-phrase, right? Otherwise this is unnecessary work, and there won't be data if it's not the line_phrase_cj file?
You can check `if (isPhraseBreaking)` before calling this function, or inside. I prefer before.

Reply:
The initJapanesePhraseParameter is called during the constructor. This object follow the singleton pattern and will only be created once (actually, twice, one for Korean case one for CJ case) and are shared by both the case the isPhraseBreaking is true or false and shared between thread. If we lazy load this part we then need to mutex it to protect thread safety. The cost of that may not be smaller than always call initJapanesePhraseParameter

Reply:
makes sense tnx